ttnn/cpp/ttnn/operations/downsample/device/downsample_op.hpp

```diff
@@ -36,6 +54,10 @@ Tensor downsample (const Tensor &a, std::array<uint32_t, 5> downsample_params, s
 // uint32_t get_num_cores(CoreCoord grid_size, uint32_t nblocks);
 // }
 
-}  // namespace tt_metal
+} // namespace data_movement
+
+}  // namespace operations
+
+constexpr auto downsample = ttnn::register_operation<ttnn::operations::data_movement::Downsample>("ttnn::downsample");
```

Comment:
You are mixing <Operation> and Execute<Operation> structures in one. Please follow the high level structure.
struct Downsample should contain all these 
```
    std::array<uint32_t, 5> downsample_params;
    DataType output_dtype;
    void validate(const std::vector<Tensor> &input_tensors) const;
    std::vector<tt::tt_metal::Shape> compute_output_shapes(const std::vector<Tensor> &input_tensors) const;
    std::vector<Tensor> create_output_tensors(const std::vector<Tensor> &input_tensors) const;
    operation::ProgramWithCallbacks create_program(const std::vector<Tensor>& input_tensors, std::vector<Tensor> &output_tensors) const;
```

but not execute_on_worker/main_thread


Please make yourself familiar with 
https://tenstorrent.github.io/tt-metal/latest/ttnn/ttnn/adding_new_ttnn_operation.html

CC @arakhmati

Reply:
Done.